src/components/Course/Course.vue

```diff
@@ -67,9 +70,20 @@ export default Vue.extend({
     };
   },
   computed: {
-    cautionString(): string | null {
-      if (this.duplicatedCourseCodeList == null) return null;
-      return this.duplicatedCourseCodeList.includes(this.courseObj.code) ? 'Duplicate' : null;
+    courseCautions(): CourseCautions {
+      return getCourseCautions(this.courseObj);
+    },
```

Comment:
Can we not move `getCourseCautions(...)` to Course.vue? Why did you keep it in CourseCaution.vue?

Reply:
@tcho6319  Now I managed to keep it within the CourseCaution completely. Now you only need to pass in the course object and it can derive all the warnings.